### PR TITLE
get_details and get_link_details return SignedHeaderHashed and ZomeInfo includes ZomeId

### PR DESCRIPTION
**Pull request by freesig**
# Breaking changes
### LinkDetails:
From:
```rust
Vec<(CreateLink, Vec<DeleteLink>)>
```
To:
```rust
Vec<(SignedHeaderHashed, Vec<SignedHeaderHashed>)>
```
### ElementDetails
From:
```rust
pub struct ElementDetails {
    pub element: Element,
    pub deletes: Vec<Delete>,
}
```
To:
```rust
pub struct ElementDetails {
    pub element: Element,
    pub deletes: Vec<SignedHeaderHashed>,
}
```
### EntryDetails
From:
```rust
pub struct EntryDetails {
    pub entry: Entry,
    pub headers: Vec<Header>,
    pub deletes: Vec<Delete>,
    pub updates: Vec<Update>,
    pub entry_dht_status: EntryDhtStatus,
}
```
To:
```rust
pub struct EntryDetails {
    pub entry: Entry,
    pub headers: Vec<SignedHeaderHashed>,
    pub deletes: Vec<SignedHeaderHashed>,
    pub updates: Vec<SignedHeaderHashed>,
    pub entry_dht_status: EntryDhtStatus,
}
```
## ZomeInfo
Now includes ZomeId so you can find out which zome position you are in.
This is useful for calling query.

### CODE REVIEW

**Comment by Connoropolous** [comment]
Looking good. It looks like you have to<>from reversed though,  in your  comment?

**Comment by freesig** [comment]
> This is great pray. It looks like you have to<>from reversed though, in your comment?

Ha this is what happens when I code too early in the morning, nice catch!

File: crates/zome_types/src/metadata.rs [hunk 1]
```diff
@@ -36,14 +32,14 @@ pub struct EntryDetails {
     pub entry: Entry,
     /// Create relationships.
```
Comment (Connoropolous) [inline crates/zome_types/src/metadata.rs:33]: Maybe this code comment could be updated with the improved description you gave me in the chat, of what these fields represent?

**Comment by freesig** [comment]
I added some better docs and included ZomeId in ZomeInfo

File: CHANGELOG-UNRELEASED.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 - Added App Validation workflow that runs app validation as authority [#330](https://github.com/holochain/holochain/pull/330)
 - BREAKING: get_details and get_links_details return SignedHeaderHashed instead of the header types [390](https://github.com/holochain/holochain/pull/390)
+- BREAKING: ZomeInfo now returns the ZomeId [390](https://github.com/holochain/holochain/pull/390)
```
Comment (Connoropolous) [inline CHANGELOG-UNRELEASED.md:10]: Is this considered "breaking" though its a new addition?

Comment (freesig) [reply CHANGELOG-UNRELEASED.md:10]: If someone is destructing the struct it will break their code

**Review by Connoropolous — APPROVED** [review]
From my reading of this, it's not just an improvement to the API but to the code. 

I have lost track of the context for adding the ZomeId to the ZomeInfo response, but I want to add, for future reference, that the changes from e.g. `Vec<Update>` to `Vec<SignedHeaderHashed>` arose from the desire to access one of such Update items from a `get_details!` call, where the response is not an ElementDetail, but an EntryDetail, meaning the related headers are not known, and the one accessing them may (as I did) to immediately following call `get!` for the header itself, which we can now access as a hash to pass that function.

**Review by thedavidmeister — APPROVED** [review]

**Pull request merged**